beacon-light-client/plonky2/crates/circuits/src/deposit_accumulator_balance_aggregator_diva/first_level.rs

```diff
@@ -0,0 +1,505 @@
+            &input.validator_gindex,
+        );
+
+        builder.assert_implication(deposit_is_real, is_valid);
+
+        assert_bool_arrays_are_equal(builder, &input.validator.pubkey, &input.deposit.pubkey);
+
+        let four = builder.constant_biguint(&BigUint::from_u64(4u64).unwrap());
+        let balance_inner_index = builder.rem_biguint(&input.validator_gindex, &four);
+        let balance = get_balance_from_leaf(builder, &input.balance_leaf, balance_inner_index);
+        let balance_gindex = builder.div_biguint(&input.validator_gindex, &four);
+
+        let is_valid = validate_merkle_proof_sha256(
+            builder,
+            &input.balance_leaf,
+            &input.balances_root,
+            &input.balance_branch,
+            &balance_gindex,
+        );
+
+        builder.assert_implication(deposit_is_real, is_valid);
+
+        let (is_non_activated, is_active, is_exited) = get_validator_status(
+            builder,
+            &input.validator.activation_epoch,
+            &input.current_epoch,
+            &input.validator.exit_epoch,
+        );
+
+        let zero_validator_status_stats: ValidatorStatusStatsTarget = builder.zero_init();
+        let mut validator_status_stats = ValidatorStatusStatsTarget {
+            non_activated_count: is_non_activated.target,
+            active_count: is_active.target,
+            exited_count: is_exited.target,
+            slashed_count: input.validator.slashed.target,
+        };
+        validator_status_stats = builder.select_target(
+            deposit_is_real,
+            &validator_status_stats,
+            &zero_validator_status_stats,
+        );
+
+        let is_relevant = get_validator_relevance(
+            builder,
+            &input.validator.activation_epoch,
+            &input.current_epoch,
+            &input.validator.withdrawable_epoch,
+        );
+
+        let zero_biguint = builder.zero_biguint();
+        let validator_balance = select_biguint(builder, is_relevant, &balance, &zero_biguint);
+
+        let zero_accumulated_data: AccumulatedDataTargetDiva = builder.zero_init();
+        let mut accumulated_data = AccumulatedDataTargetDiva {
+            balance: validator_balance,
+            validator_status_stats,
+        };
+        accumulated_data =
+            builder.select_target(deposit_is_real, &accumulated_data, &zero_accumulated_data);
+
+        Self::Target {
+            accumulated_data,
+
+            validator: input.validator,
+            commitment_mapper_root: input.commitment_mapper_root,
+            commitment_mapper_branch: input.commitment_mapper_branch,
+            validator_gindex: input.validator_gindex,
+            deposit: input.deposit,
+            balances_root: input.balances_root,
+            balance_leaf: input.balance_leaf,
+            balance_branch: input.balance_branch,
+            is_dummy: input.is_dummy,
+            current_epoch: input.current_epoch,
+            deposit_hash_tree_root,
```

Comment:
rename this to deposits_hash_tree_root as it's the hash tree root of the whole deposits tree